src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionAvoidEscapedUnicodeCharactersTest.java

```diff
@@ -0,0 +1,95 @@
+
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.checks.AvoidEscapedUnicodeCharactersCheck;
+
+public class XpathRegressionAvoidEscapedUnicodeCharactersTest extends AbstractXpathTestSupport {
+    private final String checkName = AvoidEscapedUnicodeCharactersCheck.class.getSimpleName();
+
+    @Override
+    protected String getCheckName() {
+        return checkName;
+    }
+
+    @Test
+    public void testOne() throws Exception {
+        final File fileToProcess = new File(getPath(
+                "SuppressionXpathRegressionAvoidEscapedUnicodeCharactersDefault.java"));
+
+        final DefaultConfiguration moduleConfig =
+                createModuleConfig(AvoidEscapedUnicodeCharactersCheck.class);
+
+        final String[] expectedViolation = {
+            "4:44: " + getCheckMessage(AvoidEscapedUnicodeCharactersCheck.class, MSG_KEY),
+        };
+
+        final List<String> expectedXpathQueries = Arrays.asList(
+            "/CLASS_DEF[./IDENT"
+                + "[@text='SuppressionXpathRegressionAvoidEscapedUnicodeCharactersDefault']]"
+                + "/OBJBLOCK/VARIABLE_DEF[./IDENT[@text='unitAbbrev2']]"
+                + "/ASSIGN/EXPR[./STRING_LITERAL[@text='\\u03bcs']]",
+            "/CLASS_DEF[./IDENT"
+                + "[@text='SuppressionXpathRegressionAvoidEscapedUnicodeCharactersDefault']]"
+                + "/OBJBLOCK/VARIABLE_DEF[./IDENT[@text='unitAbbrev2']]"
+                + "/ASSIGN/EXPR/STRING_LITERAL[@text='\\u03bcs']"
+        );
+
+        runVerifications(moduleConfig, fileToProcess, expectedViolation,
+                expectedXpathQueries);
+    }
+
+    @Test
+    public void testTwo() throws Exception {
+        final File fileToProcess = new File(getPath(
+                "SuppressionXpathRegressionAvoidEscapedUnicodeCharactersTwo.java"));
+
+        final DefaultConfiguration moduleConfig =
+                createModuleConfig(AvoidEscapedUnicodeCharactersCheck.class);
+        moduleConfig.addAttribute("allowEscapesForControlCharacters", "true");
+
+        final String[] expectedViolation = {
+            "4:34: " + getCheckMessage(AvoidEscapedUnicodeCharactersCheck.class, MSG_KEY),
+        };
+
+        final List<String> expectedXpathQueries = Arrays.asList(
+            "/CLASS_DEF[."
+                + "/IDENT[@text='SuppressionXpathRegressionAvoidEscapedUnicodeCharactersTwo']]"
+                + "/OBJBLOCK/VARIABLE_DEF[./IDENT[@text='unitAbbrev9']]"
+                + "/ASSIGN/EXPR[./STRING_LITERAL[@text='\\u03bcs']]",
+            "/CLASS_DEF[."
+                + "/IDENT[@text='SuppressionXpathRegressionAvoidEscapedUnicodeCharactersTwo']]"
+                + "/OBJBLOCK/VARIABLE_DEF[./IDENT[@text='unitAbbrev9']]"
+                + "/ASSIGN/EXPR/STRING_LITERAL[@text='\\u03bcs']"
+        );
+
+        runVerifications(moduleConfig, fileToProcess, expectedViolation,
+                expectedXpathQueries);
+    }
```

Comment:
there are 4 properties + default config, but only 2 tests here

Reply:
Okay, I'll add more test cases.

Reply:
Done. 
Just one thing, for the test with `allowByTailComment`, I had to write `/* warn */` at the start as the check was considering it as a tail comment and wasn't getting violated.
Also, I had one doubt. In the documentation of `AvoidEscapedUnicodeCharacters`, I think it should be `tail comment` instead of `trail comment` as shown in the code below.
```
 <p>
 * Example of using escapes with trail comment:
 * </p>
 * <pre>
 * String unitAbbrev = "&#92;u03bcs"; // Greek letter mu, "s"
 * </pre>
 * <p>An example of how to configure the check to allow using escapes
 * if trail comment is present:
 * </p>
```

Reply:
it is fine